core/error_macros.h

```diff
@@ -129,38 +131,44 @@ extern bool _err_error_exists;
 
 // (*): See https://stackoverflow.com/questions/257418/do-while-0-what-is-it-good-for
 
-#define ERR_FAIL_INDEX(m_index, m_size)                                                                                    \
-	do {                                                                                                                   \
-		if (unlikely((m_index) < 0 || (m_index) >= (m_size))) {                                                            \
-			_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Index " _STR(m_index) " out of size (" _STR(m_size) ")."); \
-			return;                                                                                                        \
-		} else                                                                                                             \
-			_err_error_exists = false;                                                                                     \
+#define ERR_FAIL_INDEX(m_index, m_size)                                                                                                  \
+	do {                                                                                                                                 \
+		if (unlikely((m_index) < 0 || (m_index) >= (m_size))) {                                                                          \
+			char err[1024];                                                                                                              \
```

Comment:
can you do it without allocating exta stack, so functions dont get this extra stack computed? just use something like
String("blsblah"+itos(size)+"blablah).utf8().get_data() to pass the error

Reply:
That was actually my first idea to implement this, but if I use String I get compile errors about not being defined and if I include ustring.h I get tons of compile errors.

Reply:
ah yeah, i can imagine the function is used in places where ustring.h is not defined.
why not calling an intermediate C function in error_macros.cpp that then calls the function to print?

Reply:
Alright, I've added a function now, looks better this way.

Reply:
Now that it's in the .cpp, you could use "ustring.h" no? I see that .cpp includes "core/io/logger.h" which uses String too.

Reply:
Sure, moved it to use String now.